models/issue.rb

```diff
@@ -86,6 +147,57 @@ def self.old_issues(repo)
       all
    end
 
+   ##
+   # Returns array of Issue objects on a milestone with a due date
+   ##
+   def self.due_issues(issues, repo)
+      Log.info "Selecting issues with deadlines"
+      all = []
+      Github.each(issues) do |issue_data|
+         issue = Issue.new(repo, issue_data)
+         next if issue.is_pull_request
+
+         unless issue.due_on.nil?
+            all << issue
+         end
+      end
+      all
+   end
+
+   ##
+   # Returns array of Issue objects with an empty description
+   ##
+   def self.empty_issues(issues, repo)
+      Log.info "Selecting issues with deadlines"
+      all = []
+      Github.each(issues) do |issue_data|
+         issue = Issue.new(repo, issue_data)
+         next if issue.is_pull_request
+
+         if issue.description.empty?
+            all << issue
+         end
+      end
+      all
+   end
+
+   ##
+   # Returns array of Issue objects which are pull requests without an
+   # associated issue
+   ##
+   def self.disconnected_pulls(pulls, repo)
+      Log.info "Selecting pulls without an pull"
+      all = []
+      Github.each(pulls) do |pull_data|
+         pull = Issue.new(repo, pull_data)
+
+         unless pull.has_issue
+            all << pull
+         end
+      end
+      all
```

Comment:
I think all these `all=[]; Github.each...` constructs could be rebuilt with [`Enumerable#reject`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-reject) and [`Enumerable#find_all`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-find_all) and be slightly more clean as a result.